### ISSUE TITLE
unreachable code error while using JSON

### ISSUE THREAD

**Issue by spino17** [JairusSW/as-json#82]
I am trying out below code:
```typescript

import { JSON } from "json-as/assembly";

@json
class TokenMetaData {
  id: u64;
  name: string;
  uri: string;

  constructor(id: u64, name: string, uri: string) {
    this.id = id;
    this.name = name;
    this.uri = uri;
  }
}

@json
class NonFungibleToken {
  owner: string;
  counter: u64;
  tokens: Map<u64, TokenMetaData>;
  owners: Map<u64, string>;
  balances: Map<string, u64[]>;

  constructor() {
    this.owner = "";
    this.counter = 0;
    this.tokens = new Map();
    this.owners = new Map();
    this.balances = new Map();
  }
}

export function mint(): void {
  let state = JSON.parse<NonFungibleToken>(
    '"{"owner":"","counter":0,"tokens":{},"owners":{},"balances":{}}"'
  );
  // let state = new NonFungibleToken();

  if (!state.balances.has("bhavya")) {
    state.balances.set("bhavya", []);
  }
}

```
Which is giving me
```
Error: RuntimeError: unreachable
```

but when I change it to
```typescript

import { JSON } from "json-as/assembly";

@json
class TokenMetaData {
  id: u64;
  name: string;
  uri: string;

  constructor(id: u64, name: string, uri: string) {
    this.id = id;
    this.name = name;
    this.uri = uri;
  }
}

@json
class NonFungibleToken {
  owner: string;
  counter: u64;
  tokens: Map<u64, TokenMetaData>;
  owners: Map<u64, string>;
  balances: Map<string, u64[]>;

  constructor() {
    this.owner = "";
    this.counter = 0;
    this.tokens = new Map();
    this.owners = new Map();
    this.balances = new Map();
  }
}

export function mint(): void {
  //let state = JSON.parse<NonFungibleToken>(
  //  '"{"owner":"","counter":0,"tokens":{},"owners":{},"balances":{}}"'
  //);
  let state = new NonFungibleToken();

  if (!state.balances.has("yamini")) {
    state.balances.set("yamini", []);
  }
}
```
it is working fine.

**Comment by JairusSW** [comment]
Thanks for the better examlle.
I'll fix this tonight, sorry

**Comment by spino17** [comment]
Thanks @JairusSW for the response. Let me know if you need anything else from my end.

**Comment by JairusSW** [comment]
Just a note, I'd reccomend using default values with `json-as`. Its designed to work well with them

**Comment by JairusSW** [comment]
Can you try now? Fix live on [json-as@0.9.9](https://npmjs.com/package/json-as)

**Comment by spino17** [comment]
yeah, now it is working perfectly! thanks @JairusSW for fixing it
